athena-elasticsearch/src/main/java/com/amazonaws/connectors/athena/elasticsearch/ElasticsearchMetadataHandler.java

```diff
@@ -140,9 +198,9 @@ public ListTablesResponse doListTables(BlockAllocator allocator, ListTablesReque
         logger.info("doListTables: enter - " + request);
 
         List<TableName> indices = new ArrayList<>();
-        String endpoint = ElasticsearchHelper.getDomainEndpoint(request.getSchemaName());
+        String endpoint = getDomainEndpoint(request.getSchemaName());
 
-        if (!endpoint.isEmpty()) {
+        if (endpoint != null) {
```

Comment:
what happens if it is indeed null? it looks like you return an empty schema, I don't think that is a good idea. I'd throw an error saying what happened (e.g. no endpoint)

Reply:
Also further down it seems you will return an empty schema if schema is null. That seems like an anti-pattern. When would schema be null? most likely we should be throwing some useful error message if that is the case.

Reply:
Added exception handling to handle endpoint related errors and throw when endpoint does not exist in the domain map.